### PR TITLE
Unexpected tabs reset. Broken label_config saving. Another small fixes

### PR DESCRIPTION
**Pull request by makseq**

### CODE REVIEW

File: label_studio/projects/api.py [hunk 1]
```diff
@@ -195,11 +195,14 @@ def delete(self, request, *args, **kwargs):
...
     def patch(self, request, *args, **kwargs):
         project = self.get_object()
-        label_config = self.request.query_params.get('label_config')
+        label_config = self.request.data.get('label_config')
 
         # config changes can break view, so we need to reset them
-        if parse_config(label_config) != parse_config(project.label_config):
-            View.objects.filter(project=project).all().delete()
+        try:
+            if label_config and parse_config(label_config) != parse_config(project.label_config):
```
Comment (niklub) [inline label_studio/projects/api.py:202]: there is another function in this file called `config_essential_data_has_changed` - it's better to modify and reuse it here

File: label_studio/tasks/models.py [hunk 2]
```diff
@@ -255,7 +255,6 @@ class Annotation(models.Model):
     objects = AnnotationManager()
     tracker = FieldTracker(fields=['ground_truth', 'result'])
 
-    state = JSONField('state', null=True, default=dict, help_text='Editor state (system data)')
```
Comment (niklub) [inline label_studio/tasks/models.py:258]: why?

File: label_studio/tasks/api.py [hunk 3]
```diff
@@ -139,38 +136,6 @@ def put(self, request, *args, **kwargs):
         return super(TaskAPI, self).put(request, *args, **kwargs)
 
 
-class TaskCancelAPI(APIView):
```
Comment (niklub) [inline label_studio/tasks/api.py:142]: hm, and how Skip is handled now?

**Review by niklub — CHANGES_REQUESTED** [review]

File: label_studio/core/old_ls_migration.py [hunk 4]
```diff
@@ -42,15 +42,15 @@ def suppress_autotime(model, fields):
 def _migrate_tasks(project_path, project):
     """ Migrate tasks from json file to database objects"""
     tasks_path = project_path / 'tasks.json'
-    with io.open(os.path.abspath(tasks_path)) as t:
+    with io.open(os.path.abspath(tasks_path), encoding='utf-8') as t:
```
Comment (niklub) [inline label_studio/core/old_ls_migration.py:45]: do we have a github issue patched by this fix?

File: label_studio/tasks/models.py (hunk 2 above)
Comment (makseq) [reply label_studio/tasks/models.py:258]: Because we don't need it for more over a year

File: label_studio/tasks/api.py (hunk 3 above)
Comment (makseq) [reply label_studio/tasks/api.py:142]: AnnotationListAPI

File: label_studio/core/old_ls_migration.py (hunk 4 above)
Comment (makseq) [reply label_studio/core/old_ls_migration.py:45]: I didn't see it, but old projects with non utf8 chars won't be migrated

File: label_studio/tasks/models.py (hunk 2 above)
Comment (niklub) [reply label_studio/tasks/models.py:258]: I meant that we can remove later all useless stuff in one migration, no rush for that now

**Review by niklub — APPROVED** [review]

**Pull request merged**